[OpenID] Error in log at server startup when oidc is not activated

**Environment** : 7.0.0-20250116
**Scenario** :

- start a fresh server.
- display login page

**Current Behaviour :**
Error in logs :

> Caused by: java.lang.NullPointerException: Cannot invoke "io.meeds.oauth.spi.OAuthProviderType.getOauthProviderProcessor()" because the return value of "io.meeds.oauth.spi.OAuthProviderTypeRegistry.getOAuthProvider(String, java.lang.Class)" is null at io.meeds.oauth.web.openid.OpenIdAuthenticationFilter.initImpl(OpenIdAuthenticationFilter.java:36) at org.gatein.sso.agent.filter.api.AbstractSSOInterceptor.initWithParams(AbstractSSOInterceptor.java:69) at org.gatein.sso.integration.SSOFilterIntegratorPlugin.(SSOFilterIntegratorPlugin.java:79) at io.meeds.oauth.webapi.OAuthFilterIntegratorPlugin.(OAuthFilterIntegratorPlugin.java:31) at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ... 58 common frames omitted

**Expected** : no error

--- a/component/oauth-auth/src/main/java/io/meeds/oauth/web/openid/OpenIdAuthenticationFilter.java
+++ b/component/oauth-auth/src/main/java/io/meeds/oauth/web/openid/OpenIdAuthenticationFilter.java
@@ -45,11 +45,21 @@ public class OpenIdAuthenticationFilter extends AbstractSSOInterceptor {
   private static Log log = ExoLogger.getLogger(OpenIdAuthenticationFilter.class);
   private AuthenticationRegistry authenticationRegistry;
   private OpenIdProcessor openIdProcessor;
+
+  private boolean openIdEnabled;
   @Override
   protected void initImpl() {
-    authenticationRegistry = getExoContainer().getComponentInstanceOfType(AuthenticationRegistry.class);
-    OAuthProviderTypeRegistry oAuthProviderTypeRegistry = getExoContainer().getComponentInstanceOfType(OAuthProviderTypeRegistry.class);
-    openIdProcessor = (OpenIdProcessor) oAuthProviderTypeRegistry.getOAuthProvider(OAuthConstants.OAUTH_PROVIDER_KEY_OPEN_ID, OpenIdAccessTokenContext.class).getOauthProviderProcessor();
+    this.openIdEnabled = Boolean.parseBoolean(System.getProperty("exo.oauth.openid.enabled"));
+    if (this.openIdEnabled) {
+      authenticationRegistry = getExoContainer().getComponentInstanceOfType(AuthenticationRegistry.class);
+      OAuthProviderTypeRegistry
+          oAuthProviderTypeRegistry =
+          getExoContainer().getComponentInstanceOfType(OAuthProviderTypeRegistry.class);
+      openIdProcessor =
+          (OpenIdProcessor) oAuthProviderTypeRegistry.getOAuthProvider(OAuthConstants.OAUTH_PROVIDER_KEY_OPEN_ID,
+                                                                       OpenIdAccessTokenContext.class)
+                                                     .getOauthProviderProcessor();
+    }
   }
 
   @Override
@@ -60,7 +70,7 @@ public class OpenIdAuthenticationFilter extends AbstractSSOInterceptor {
     HttpServletResponse response = (HttpServletResponse) servletResponse;
 
     // Simply continue with request if we are already authenticated
-    if (request.getRemoteUser() != null) {
+    if (!this.openIdEnabled || request.getRemoteUser() != null) {
       filterChain.doFilter(request, response);
       return;
     }

